src/reports.py

```diff
@@ -0,0 +1,200 @@
+        folder: save folder for output file 
+    """
+    fig, ax = plt.subplots(2,2, figsize=(12,8))
+    top = settings["top"]
+
+    _fill_top_runners_graph(ax[0][0], "All-time" , data, settings, club,  0)
+    _fill_top_runners_graph(ax[0][1], "Last year" , data, settings, club, 52)
+    _fill_top_runners_graph(ax[1][0], "Last half year" , data, settings, club,  26)
+    _fill_top_runners_graph(ax[1][1], "Last 15 runs" , data, settings, club, 15)
+
+    fig.suptitle(f'Top {top} runners in {club} [{"Active runners and clubs" if settings["filter"] else "All data"}]' )
+    fig.tight_layout(pad=2, w_pad=2, h_pad=3)
+    fig.savefig(os.path.join(folder, f"top_runners_{club}.png"), dpi=600)
+
+
+def _longest_streaks_graph(club: str, data: dict, settings: dict, folder: str):
+    """Creates graph for longest streaks
+
+    Args:
+        club: particular club
+        data: to be processed
+        settings: report settings
+        folder: save folder for output file 
+    """
+    fig, ax = plt.subplots(figsize=(8,8))
+    axis : Axes = ax
+    top = settings["top"]
+    
+    clubstreaks = longest_streaks(data, top)[club]
+    axis.barh([nick for nick in clubstreaks], [amount for amount in clubstreaks.values()], color=settings["colors"][club])
+    axis.set_xticks(range(1, max(clubstreaks.values())+1))
+    axis.set_xlabel("Streak length")
+    fig.suptitle(f'Top {top} streaks in {club} [{"Active runners and clubs" if settings["filter"] else "All data"}]' )
+    fig.tight_layout(pad=2)
+    fig.savefig(os.path.join(folder, f"longest_streaks_{club}.png"), dpi=600)
+
+
+def club_reports(club: str, data: dict, settings: dict, savefolder: str):
+    """Manages report creation for a club
+
+    Pre-filters data based on settings
+
+    Args:
+        club: particular club
+        data: to be processed
+        settings: report settings
+        savefolder: a folder to save the reports into
+    """
+
+    filterdata = filter_data(data) if settings["filter"] else data
+    if len(top_runners(filterdata, top=0)[club]) > 0: 
+
+        if settings["graph"]:
+            if not os.path.exists(savefolder) : os.makedirs(savefolder)
+            _top_runners_graph(club, filterdata, settings, savefolder)
+            _longest_streaks_graph(club, filterdata, settings, savefolder)
+
+        print(f"reports saved: {savefolder}")
+
+    else:
+        print(f"Not enough data for {club}. Please run more ;)")
+
+
+def pr_report(data: dict, settings: dict, savefolder: str):
+    """Generates and saves a report for personal records
+
+    Pre-filters data based on settings
+
+    Args:
+        data: to be processed
+        settings: report settings
+        savefolder: a folder to save the reports into
+    """
+    filterdata = filter_data(data) if settings["filter"] else data    
+    top = settings["top"]
+    topprs = top_prs(filterdata, settings["top"])
+
+    num = len(topprs)
+    if num > 0:
+        if settings["graph"]:
```

Comment:
Érdemes előrevenni az elágazás rövidebb ágát. Itt ráadásul abból lehet return-ölni is, és akkor egy tabnyi mélységet megspórolsz:
```suggestion
    if not topprs:
        print("No personal records to show.")
        return
    num = len(topprs)
    if settings["graph"]:
        ...
```